external/diagnostics_utils/diagnostics_utils/compute_metrics.py

```diff
@@ -0,0 +1,223 @@
+from diagnostics_utils.registry import Registry
+import diagnostics_utils.transform as transform
+
+import logging
+from typing import Mapping, Sequence, Tuple
+
+import xarray as xr
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+HORIZONTAL_DIMS = ["x", "y", "tile"]
+
+
+def _prepare_diag_dict(suffix: str, ds: xr.Dataset) -> Mapping[str, xr.DataArray]:
+    """
+    Take a diagnostic dataset and add a suffix to all variable names and return as dict.
+    """
+
+    diags = {}
+    for variable in ds:
+        lower = variable.lower()
+        da = ds[variable]
+        diags[f"{lower}_{suffix}"] = da
+
+    return diags
+
+
+def merge_metrics(metrics: Sequence[Tuple[str, Mapping[str, xr.DataArray]]]):
+    out = {}
+    for name, ds in metrics:
+        out.update(_prepare_diag_dict(name, ds))
+    return xr.Dataset(out)
+
+
+metrics_registry = Registry(merge_metrics)
+
+
+def compute_metrics(
+    prediction: xr.Dataset,
+    target: xr.Dataset,
+    grid: xr.Dataset,
+    delp: xr.DataArray,
+    n_jobs: int = -1,
+):
+    return metrics_registry.compute(prediction, target, grid, delp, n_jobs=n_jobs)
+
+
+def mse(x, y, w, dims):
+    with xr.set_options(keep_attrs=True):
+        return ((x - y) ** 2 * w).sum(dims) / w.sum(dims)
+
+
+def bias(truth, prediction, w, dims):
+    with xr.set_options(keep_attrs=True):
+        return prediction - truth
+
+
+def weighted_mean(ds, weights, dims):
+    with xr.set_options(keep_attrs=True):
+        return (ds * weights).sum(dims) / weights.sum(dims)
+
+
+def zonal_mean(
+    ds: xr.Dataset, latitude: xr.DataArray, bins=np.arange(-90, 91, 2)
+) -> xr.Dataset:
+    with xr.set_options(keep_attrs=True):
+        zm = (
+            ds.groupby_bins(latitude, bins=bins)
+            .mean(skipna=True)
+            .rename(lat_bins="latitude")
+        )
+    latitude_midpoints = [x.item().mid for x in zm["latitude"]]
+    return zm.assign_coords(latitude=latitude_midpoints)
+
+
+for mask_type in ["global", "sea", "land"]:
+
+    @metrics_registry.register(f"mse_2d_{mask_type}")
+    @transform.apply("select_2d_variables")
```

Comment:
I realize this pattern is in the prognostic run, but the approach of "registering" a function with its  name as a string and then referring to it later with this same string later is circuitous. A python module is already a "registry" of functions. We can refer to a function using its name `transforms.<name>` and call it later on using normal python syntax.

Can you modify `transform.apply` to take in a function instead and modify its usage throughout?
```suggestion
    @transform.apply(transform.select_2d_variables)
```